Framework/GenAIFramework/Editor/Scripts/scene_generation/prefabs_manager.py

```diff
@@ -0,0 +1,421 @@
+"""
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+"""
+
+from pathlib import Path
+import yaml
+import xml.etree.ElementTree as ET
+from xml.dom.minidom import parseString
+import numpy as np
+from utils import (
+    parse_vector,
+    prefab_dicts_to_xml,
+)
+from prefab import AvailablePrefab, SpawnedPrefab, get_overlapping_prefabs_pairs, EditorEntityId
+import azlmbr
+
+
+class SpawnedPrefabsDict(dict):
+    def __getitem__(self, key: EditorEntityId | int) -> SpawnedPrefab:
+        key = int(key)
+        return super().__getitem__(key)
+
+    def __setitem__(self, key: EditorEntityId | int, value: SpawnedPrefab):
+        key = int(key)
+        super().__setitem__(key, value)
+
+
+class OverlapException(Exception):
+    pass
+
+
+class PrefabsManager:
+    def __init__(
+        self,
+        prefabs_info_yaml_path: None | Path = None,
+    ) -> None:
+        self.prefabs_info_yaml_path = prefabs_info_yaml_path
+        self.automatically_removed_prefabs = set()
+        self.actions_log = []
+        self.available_prefabs = {}
```

Comment:
Why not initialize it immediately? Also provides automatic type inference.
```suggestion
        self.available_prefabs = self.get_available_prefabs()
```

Reply:
We read available prefabs from yaml on each request anyway to allow modyfing prefabs without recreating the chat window.